miner/miner.go

```diff
@@ -187,6 +187,11 @@ func (self *Miner) Pending() (*types.Block, *state.StateDB) {
 	return self.worker.pending()
 }
 
+// PendingBlock returns the currently pending block.
```

Comment:
Could you please add a note to the method description, along the lines: "Note, to access both the pending block and the pending state simultaneously, please use `Pending()`, as the pending state can change between multiple method calls".

The reason I'd like to have this mentioned is because there was an ugly bug way back where the pending block and pending state had two separate methods to access and we didn't know why the API returned weird results from time to time. Turned out that sometimes the API retrieving the 2 pending "pieces" separately sometimes caught the miner updating them, so the block was out of sync with the state.